src/Cooked/Skeleton.hs

```diff
@@ -154,15 +170,28 @@ data BalanceOutputPolicy
 instance Default BalanceOutputPolicy where
   def = AdjustExistingOutput
 
--- | Which wallet to use to provide outputs for balancing and collaterals.
--- Either the first signer or an explicit wallet. In the second case, this
--- wallet must be a signer of the transaction.
-data BalancingWallet
+-- | Which UTxOs to use when balancing
+data BalancingUtxos
+  = -- | Use all UTxOs containing only a Value (no datum, no staking credential,
+    -- and no reference script) belonging to the balancing wallet.
+    BalancingUtxosAutomatic
+  | -- | Use the provided UTxOs. UTxOs belonging to scripts will be filtered out
+    BalancingUtxosWith (Set Api.TxOutRef)
+  deriving (Eq, Ord, Show)
+
+instance Default BalancingUtxos where
+  def = BalancingUtxosAutomatic
+
+-- | Whether to balance the transaction or not, and which wallet to use to
+-- provide outputs for balancing. Either the first signer or an explicit
+-- wallet. In the second case, this wallet must be a signer of the transaction.
+data BalancingPolicy
   = BalanceWithFirstSigner
   | BalanceWith Wallet
```

Comment:
Most of the current Api uses the typeclass `ToPubKeyHash` when a pkh is needed, and most of the time, users will use the `Wallet` instance of this class. In practice this means that users will almost always use wallets instead of other more concrete elements. I think having this option depend on wallet makes sense for the overall consistency and usability of cooked.